metricbeat/module/system/diskio/diskstat_windows_helper.go

```diff
@@ -0,0 +1,169 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build windows
+
+package diskio
+
+import (
+	"syscall"
+	"unsafe"
+
+	"github.com/shirou/gopsutil/disk"
+)
+
+const (
+	fileDeviceDisk               = 0x00000007
+	fileAnyAccess                = 0x0000
+	errorSuccess   syscall.Errno = 0
+)
+
+var (
+	// ioctlDiskPerformance :control code - https://docs.microsoft.com/en-us/windows/desktop/api/winioctl/ni-winioctl-ioctl_disk_performance
+	ioctlDiskPerformance        = getCtlCode(fileDeviceDisk, 0x0008, 0, fileAnyAccess)
+	modkernel32                 = syscall.NewLazyDLL("kernel32.dll")
+	procGetLogicalDriveStringsW = modkernel32.NewProc("GetLogicalDriveStringsW")
```

Comment:
`GetLogicalDriveStrings` would ideally get added at https://github.com/elastic/go-windows/blob/bb1581babc04d5cb29a2bfa7a9ac6781c730c8dd/kernel32.go#L31 and then we'd let the tooling generate the boilerplate code that wraps the method (via running [`go generate`](https://github.com/elastic/go-windows/blob/a4ab469ab3d6f167cba8cb515c3273c6b56f9fc3/doc.go#L23)).

We already did this in [gosigar](https://github.com/elastic/gosigar/blob/7bed2391b1d10309feac210aee9df44e8db2d060/sys/windows/syscall_windows.go#L136-L155) but we're trying to migrate this over to go-windows as needed.

Reply:
will follow up on that with a new PR